runtime/src/main/scala/akka/grpc/javadsl/GoogleProtobufSerializer.scala

```diff
@@ -7,13 +7,18 @@ package akka.grpc.javadsl
 import akka.annotation.ApiMayChange
 import akka.grpc.ProtobufSerializer
 import akka.util.ByteString
+import com.google.protobuf.Parser
 
 @ApiMayChange
-class GoogleProtobufSerializer[T <: com.google.protobuf.Message](clazz: Class[T]) extends ProtobufSerializer[T] {
+class GoogleProtobufSerializer[T <: com.google.protobuf.Message](parser: Parser[T]) extends ProtobufSerializer[T] {
+
+  // For binary compatibility in generated sources, can be dropped in version 2.x
+  def this(clazz: Class[T]) =
+    this(clazz.getMethod("parser").invoke(clazz).asInstanceOf[Parser[T]])
+
   override def serialize(t: T): ByteString =
     ByteString.fromArrayUnsafe(t.toByteArray)
   override def deserialize(bytes: ByteString): T = {
-    val parser = clazz.getMethod("parseFrom", classOf[Array[Byte]])
-    parser.invoke(clazz, bytes.toArray).asInstanceOf[T]
+    parser.parseFrom(bytes.toArray)
```

Comment:
Wouldn't this avoid a copy?

```suggestion
    parser.parseFrom(bytes.iterator.asInputStream)
```

Reply:
I guess one could also do:

```suggestion
    parser.parseFrom(bytes.compact.asByteBuffer)
```

But that might entail copying for fragmented ByteStrings.

Reply:
This PR is just aiming at removing the reflection, and then allow those changes since we will have the parser available :)

But yes both options might be more performant, it's really hard to know the impact without proper benchmarking as a single copy to an array can be faster than many small ones using a stream. My gut-feeling is that for small messages, copying  once into an array is faster, but it depends a lot on the implementation (reader-side, e.g. with buffer sizes), and on the fragmentation. We should probably align the implementations between Java/Scala APIs, see `ScalapbProtobufSerializer` that uses the `bytes.asByteBuffer` approach. Also related to #1163 for a benchmark on that point.

I might do such benchmark if I find the time, I think the hard part is to find the correct dimensions and sample data to show the tradeoffs, as shown in #1180 by @raboof and @ignasi35.

Reply:
I'll defer to @raboof and @ignasi35  :)